### PR TITLE
Return 404 instead of error when commit not exist

### LINKED ISSUE #31976: API /repos/{owner}/{repo}/commits call error 500 on non-existent branch
### Description

_No response_

### Gitea Version

1.23.0+dev-460-ga366af4b7a (or 1.22)

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

This works,

```
curl -X 'GET' \
  'https://demo.gitea.com/api/v1/repos/bartvdbraak/blender/commits?sha=before-big-change&stat=false&verification=false&files=false&page=1' \
  -H 'accept: application/json'
```

but changing the sha= parameter to invalid one (eg. a branch that doesn't exist)

```
curl -X 'GET' \
  'https://demo.gitea.com/api/v1/repos/bartvdbraak/blender/commits?sha=not-before-big-change&stat=false&verification=false&files=false&page=1' \
  -H 'accept: application/json'
```

results in error 500. Expectation is we get an empty result set with result code of 200 if repository exists but branch/commit id does not.

### Database

None

### PR DESCRIPTION
**Pull request by AdamMajer**
Fix #31976

### CODE REVIEW

File: routers/api/v1/repo/commits.go [hunk 1]
```diff
@@ -195,6 +197,10 @@ func GetAllCommits(ctx *context.APIContext) {
 			// get commit specified by sha
 			baseCommit, err = ctx.Repo.GitRepo.GetCommit(sha)
 			if err != nil {
+				if git.IsErrNotExist(err) {
+					ctx.Status(http.StatusNoContent)
+					return
+				}
 				ctx.Error(http.StatusInternalServerError, "GetCommit", err)
 				return
```
Comment (delvh) [inline routers/api/v1/repo/commits.go]: Two points:
1. > no commit found

Sounds like it should be `404` to me, not `204`
2. In that case, you can simplify this block to `ctx.NotFoundOrServerError`

Comment (lafriks) [reply routers/api/v1/repo/commits.go]: Yes, it should be 404 as it's no different than other endpoints return for not found user etc

**Review by lafriks — CHANGES_REQUESTED** [review]
As per comment

File: routers/api/v1/repo/commits.go (hunk 1 above)
Comment (AdamMajer) [reply routers/api/v1/repo/commits.go]: 404 is return code when repository doesn't exist. But I can live with 404 too.

**Review by lafriks — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**